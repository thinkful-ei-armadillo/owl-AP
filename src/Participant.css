.session-participant {
  display: flex;
  margin-top: 15px;
}

.session-participant img {
  display: inline-block;
  width: 30px;
  height: 30px;
  border-radius: 100%;
  font-size: 10px;
}

.participant-info {
  margin-left: 10px;
  padding-top: 5px;
}

.participant-info h6 {
  position: relative;
  color: #404853;
  margin-bottom: 0;
  margin-top: 0px;
}

.participant-status {
  padding-top: 10px;
  font-weight: 400;
  font-size: 16px;
  line-height: 30px;
  color: #8e939a;
  padding-left: 20px;
  position: relative;
  top: -10px;
}
.participant-status:after {
  content: '';
  position: absolute;
  left: 0;
  top: 20px;
  width: 10px;
  height: 10px;
  border-radius: 100%;
}

.status_green:after {
  background-color: #64d481;
}

.status_grey:after {
  background-color: #d0d2d5;
}
