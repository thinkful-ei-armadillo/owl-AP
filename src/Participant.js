import React from 'react';

export default function Participant(props){
    return (
        
    );
}