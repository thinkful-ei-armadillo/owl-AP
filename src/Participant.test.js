import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Participant from './Participant';
import store from './store';

describe('Participant component testing', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');

    //render the component, this is the actual test, if something is wrong it will fail here
    ReactDOM.render(
      <Participant
        key={1}
        name="Ronnie"
        avatar="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1'"
        status="stage"
      />,
      div
    );

    //clean up code
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <Participant
          key={1}
          name="Ronnie"
          avatar="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1'"
          status="stage"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
