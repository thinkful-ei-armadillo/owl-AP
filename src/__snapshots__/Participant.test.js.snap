// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participant component testing renders the UI as expected 1`] = `
<li
  className="session-participant"
>
  <img
    alt="participant of {altText}"
    src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1'"
  />
  <div
    className="participant-info"
  >
    <h6>
      Ronnie
    </h6>
    <div
      className="participant-status status_green"
    >
      on stage
      ;
    </div>
  </div>
</li>
`;
